src/renderer/src/state.ts

```diff
@@ -17,12 +17,22 @@ export const screenState = atom<'Chat' | 'Peers'>({
   default: 'Chat',
 })
 
+export const localPeerState = atom({
+  key: 'localPeer',
+  default: { name: 'unknown' },
+  effects: [
+    syncEffect({
+      storeKey: 'dataChannel',
+      refine: object({ name: string() }),
+    }),
+  ],
+})
```

Comment:
Here we set up an atom state that is automatically synced from the main process.  By setting up a store named `dataChannel` at the root of the react app and setting up the data-type using `refine`, this value is updated from the backend every where this atom is referenced in the renderer.